bundles/org.openhab.core.io.rest.sitemap/src/main/java/org/openhab/core/io/rest/sitemap/internal/SitemapResource.java

```diff
@@ -333,6 +375,28 @@ public Object createEventSubscription() {
         return JerseyResponseBuilderUtils.created(uri.toASCIIString());
     }
 
+    /**
+     *
+     * Subscribe to a whole sitemap (all pages at once).
+     * This is not a recommended option as it incurs high SSE traffic
+     * and might pose a high load on the server, depending on the sitemap size.
+     * No built-in openhab UI should use this functionality.
+     *
+     */
+    @GET
+    @Path(SEGMENT_EVENTS + "/{subscriptionid: [a-zA-Z_0-9-]+}/*")
+    @Produces(MediaType.SERVER_SENT_EVENTS)
+    @Operation(operationId = "getSitemapEvents", summary = "Get sitemap events for a whole sitemap. Not recommended due to potentially high traffic.", responses = {
+            @ApiResponse(responseCode = "200", description = "OK"),
+            @ApiResponse(responseCode = "400", description = "Page not linked to the subscription."),
```

Comment:
Please update the description.
I have even the felling this response can not occur; if that is true, rather remove this line.

Reply:
The method `getSitemapEvents` returns a `BAD_REQUEST` when there is either no sitemap or no pageId (when trying to subscribe to a page only) or when either of them could not be set as location for the subscription